Makefile

```diff
@@ -7,7 +7,7 @@ OBJDIR ?= $(dir $(OUT))obj
 CXXFLAGS = -Wall -Wextra -Werror -std=c++17 -O2 -g -fPIC -DONNX_NAMESPACE=onnx
 LIBS = -lpoplar -lpopart -lpopops -lpopsparse -lpoputil
 
-OBJECTS = $(OBJDIR)/static_spmm.o
+OBJECTS = $(OBJDIR)/static_spmm.o $(OBJDIR)/replicatedallreducetp.o
```

Comment:
Shall we just drop the `tp`? If we're able to use `autograd_proxy` to remove the backwards or forwards pass at will, perhaps we can eventually just make it a 100% plain replicated all reduce.

Reply:
(But then do we lose the custom op C++ & just wrap the existing one?)

Reply:
This might be worth exploring. I'm keen to hear what @thecharlieblake  thinks.

Reply:
I think the issue is that you need a forward function to pass to autograd_proxy that inserts `allreduce` in the backward pass, e.g., `copy_with_allreduce_grad`. This would still require a custom op.